native_client/c_speech_features/c_speech_features.h

```diff
@@ -0,0 +1,222 @@
+
+/**
+ * @brief Apply a cepstral lifter on a matrix of cepstra.
+ *
+ * Apply a cepstral lifter on a matrix of cepstra. This has the effect of
+ * increasing the magnitude of high-frequency DCT coefficients.
+ *
+ * @param aCepstra The matrix of mel-cepstra.
+ * @param aNFrames The number of frames.
+ * @param aNCep The number of cepstra per frame.
+ * @param aCepLifter The lifting coefficient to use. 0 disables lifting.
+ *                   (e.g. 22)
+ */
+void csf_lifter(float** aCepstra,
+                int aNFrames,
+                int aNCep,
+                int aCepLifter);
+
+/**
+ * @brief Perform preemphasis on an input signal.
+ *
+ * Perform preemphasis on an input signal.
+ *
+ * @param aSignal The signal to filter.
+ * @param aSignalLen The length of the signal array.
+ * @param aCoeff The preemphasis coefficient. 0 is no filter. (e.g. 0.95)
+ *
+ * @return The filtered signal. The user is responsible for freeing this array.
+ */
+float* csf_preemphasis(const short* aSignal,
+                       unsigned int aSignalLen,
+                       float aCoeff);
+
+/**
+ * @brief Frame a signal into overlapping frames.
+ *
+ * Frame a signal into overlapping frames.
+ *
+ * @param aSignal The signal to frame.
+ * @param aSignalLen The length of the signal array.
+ * @param aFrameLen The length of each frame in samples.
+ * @param aPaddedFrameLen If greater than @p aFrameLen, @p aPaddedFrameLen -
+ *                        @p aFrameLen zeros will be appended to each frame.
+ * @param aFrameStep The number of samples after the start of the previous frame
+ *                   that the next frame should begin.
+ * @param[out] aFrames An array of frames, of shape (@c frames, @p aFrameLen).
+ *                     The user is responsible for freeing each row in this
+ *                     array, as well as the array itself.
+ *
+ * @return The number of frames.
+ */
+int csf_framesig(const float* aSignal,
+                 unsigned int aSignalLen,
+                 int aFrameLen,
+                 int aPaddedFrameLen,
+                 int aFrameStep,
+                 float*** aFrames);
+
+/**
+ * @brief Compute the magnitude spectrum of frames.
+ *
+ * Compute the magnitude spectrum of each frame in frames. If frames is an
+ * NxD matrix, output will be Nx@p aNFFT.
+ *
+ * @param aFrames The array of frames.
+ * @param aNFrames The number of frames.
+ * @param aNFFT The FFT length to use.
+ *
+ * @return An array containing the magnitude spectrum of the
+ *         corresponding frame, of shape (@p aNFrames, @p aNFFT / 2 + 1). The
+ *         user is responsible for freeing each row in this array, as well as
+ *         the array itself.
+ */
+float** csf_magspec(const float** aFrames,
+                    int aNFrames,
+                    int aNFFT);
+
+/**
+ * @brief Compute the power spectrum of frames.
+ *
+ * Compute the power spectrum of each frame in frames. If frames is an
+ * NxD matrix, output will be Nx@p aNFFT.
```

Comment:
Yes, you're right, I misread my code :p Will just remove the comment, it's needless duplication anyway.